tools/inference_gradio.py

```diff
@@ -0,0 +1,126 @@
+"""Anomalib Gradio Script.
+
+This script provide a gradio web interface
+"""
+
+from argparse import ArgumentParser, Namespace
+from importlib import import_module
+from pathlib import Path
+from typing import Tuple, Union
+
+import gradio as gr
+import gradio.inputs
+import gradio.outputs
+import numpy as np
+from omegaconf import DictConfig, ListConfig
+from skimage.segmentation import mark_boundaries
+
+from anomalib.config import get_configurable_parameters
+from anomalib.deploy.inferencers.base import Inferencer
+from anomalib.post_processing import compute_mask, superimpose_anomaly_map
+
+
+def infer(
+    image: np.ndarray, inferencer: Inferencer, threshold: float = 50.0
+) -> Tuple[np.ndarray, float, np.ndarray, np.ndarray, np.ndarray]:
+    """Inference function, return anomaly map, score, heat map, prediction mask ans visualisation.
+
+    Args:
+        image (np.ndarray): image to compute
+        inferencer (Inferencer): model inferencer
+        threshold (float, optional): threshold between 0 and 100. Defaults to 50.0.
+
+    Returns:
+        Tuple[np.ndarray, float, np.ndarray, np.ndarray, np.ndarray]:
+        anomaly_map, anomaly_score, heat_map, pred_mask, vis_img
+    """
+    # Perform inference for the given image.
+    threshold = threshold / 100
+    anomaly_map, anomaly_score = inferencer.predict(image=image, superimpose=False)
+    heat_map = superimpose_anomaly_map(anomaly_map, image)
+    pred_mask = compute_mask(anomaly_map, threshold)
+    vis_img = mark_boundaries(image, pred_mask, color=(1, 0, 0), mode="thick")
+    return anomaly_map, anomaly_score, heat_map, pred_mask, vis_img
+
+
+def get_args() -> Namespace:
+    """Get command line arguments.
+
+    Returns:
+        Namespace: List of arguments.
+    """
+    parser = ArgumentParser()
+    parser.add_argument("--config", type=Path, required=True, help="Path to a model config file")
+    parser.add_argument("--weight_path", type=Path, required=True, help="Path to a model weights")
+    parser.add_argument("--meta_data", type=Path, required=False, help="Path to JSON file containing the metadata.")
+
+    parser.add_argument(
+        "--threshold",
+        type=float,
+        required=False,
+        default=75.0,
+        help="Value to threshold anomaly scores into 0-100 range",
+    )
+
+    parser.add_argument("--share", type=bool, required=False, default=False, help="Share Gradio `share_url`")
+
+    args = parser.parse_args()
+
+    return args
+
+
+def get_inferencer(gladio_args: Union[DictConfig, ListConfig]) -> Inferencer:
```

Comment:
I guess `gladio_args` should be `gradio_args`?

Reply:
I like the idea to create `get_inferencer` function. It would be nice if the function args would be explicit rather than passing args all together. For example, something like:

```
def get_inferencer(config_path: Path, weight_path: Path, meta_data_path: Path) -> Inferencer:
```

This way, we could even use this for `inference.py` (In a separate PR, of course :) )

Reply:
Yes gradio of course.

About the `get_inferencer`, yes if used out of the scope of gradio it may be usefull